hadoop-ozone/dev-support/checks/junit.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -u -o pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$DIR/../../.." || exit 1
+
+: ${CHECK:="unit"}
+: ${ITERATIONS:="1"}
+
+export MAVEN_OPTS="-Xmx4096m"
+MAVEN_OPTIONS='-B -Dskip.npx -Dskip.installnpx'
+mvn ${MAVEN_OPTIONS} -DskipTests clean install
+
+REPORT_DIR=${OUTPUT_DIR:-"$DIR/../../../target/${CHECK}"}
+mkdir -p "$REPORT_DIR"
+
+rc=0
+for i in $(seq 1 ${ITERATIONS}); do
+  if [[ ${ITERATIONS} -gt 1 ]]; then
+    original_report_dir="${REPORT_DIR}"
+    REPORT_DIR="${original_report_dir}/iteration${i}"
+    mkdir -p "${REPORT_DIR}"
+  fi
+
+  mvn ${MAVEN_OPTIONS} -fae "$@" test \
+    | tee "${REPORT_DIR}/output.log"
+  irc=$?
+
+  # shellcheck source=hadoop-ozone/dev-support/checks/_mvn_unit_report.sh
+  source "${DIR}/_mvn_unit_report.sh"
+
+  if [[ ${ITERATIONS} -gt 1 ]]; then
+    REPORT_DIR="${original_report_dir}"
+    echo "Iteration ${i} exit code: ${irc}" | tee -a "${REPORT_DIR}/output.log"
```

Comment:
```suggestion
    echo "Iteration ${i} exit code: ${irc}" | tee -a "${REPORT_DIR}/summary.txt"
```

Based on the existing convention, output.log contains the full std output, summary.txt the information.

Reply:
I've updated the patch based on this suggestion.  Now that I see the results of a repeated run, a question occurred to me: should we print only failed iterations (to match how we only list failed tests in summary) or all of them?

Reply:
When we had the Kubernetes based build / notification logic, it worked based on the number of lines in the summary.txt. Today we have no such dependencies. I am fine with keeping it simple in this form (include all lines).